### ISSUE TITLE
Add support for whitelist filtering based on operation type

### ISSUE THREAD

**Issue by eperott** [Ericsson/ecaudit#29]
Currently ecAudit support filtering based on user/resource.

We should add support for whitelist filters based on type of operation. This is mostly useful for filtering on data, like having the ability to log writes, but not queries. Even though this may be less useful for other resources (like roles and schema changes), the solution must provide a reasonable solution for all cases.

In Cassandra 4.0 there is a similar filtering mechanism based on "categories" which in turn are derived from "types" (i.e type of operation).

**Comment by eperott** [comment]
A lot of the improvements which has been made while implementing this ticket are now merged as separate patches. This should make the resolution of this ticket smaller and more to the point.

This ticket will extend the public API in that it will allow the user to whitelist specific operations. But it will also break backwards compatibility in a couple of ways:
- Representation on disk will change since specific operations (SELECT, MODIFY etc) will be part of the stored whitelist filter.
- When user list a role the result will now contain specific operations, so scripts which depend on the output may break.

In short, this will trigger a major version change. And so while we're at it i propose to change some other things as well.
- Disable ability to set whitelist at the time when a role is created. That way, the only way to update a whitelist would be via the "ALTER ROLE" statement.
- Limit supported options when changing whitelist such that the user can specify _one_ operation and _one_ resource at a time. This would make it impossible to grant a user whitelist on EXECUTE for both a 
`function` and `connections` at the same time. But that doesn't make much sense any way.

**Comment by eperott** [comment]
PR created with most things in place. Some things remain to close this up.
Need to make final decision on following and implement:
- New structure of whitelist table?
- Create new table or re-use old?
- Clean-up by custom option, or altering system_schema tables manually?
